test/types/schema_helpers.test-d.ts

```diff
@@ -127,6 +135,41 @@ expectAssignable<SchemaWithIdNumberInterface | null>(
   (await interfaceNumberTestCollection.findOneAndUpdate({ a: 1 }, { a: 5 })).value
 );
 
+expectAssignable<SchemaWithIdNumberType | null>(
```

Comment:
Without helpful errors / describe blocks / test names, it's hard to GROK what these tests are testing.  I think they're testing the return type of `findOneAnd*` methods when `includeResultMetadata` is false?

1. should these tests live in findX.test-d.ts?  They're not actually related to how we infer the collection schema (which is that this file is for I think)
2. can we comment them to make it clear what they're testing?  Something I've considered in the past is an anonmyous scope with a comment:
```typescript
// findOneAnd* methods return the modified document when `includeResultMetadata` is false
{ 
	expectAssignable<SchemaWithIdNumberInterface | null>(
  		await interfaceNumberTestCollection.findOneAndDelete({ a: 1 }, { includeResultMetadata: false })
	);
   
	...
} 
```

Reply:
I've reverted the schema_helpers.test-d.ts changes per your comment, and simplified type testing by just adding 3 tests (1 for each method) in findX-test-d.ts and added a comment to them.